webrtc.html

```diff
@@ -4138,90 +4260,123 @@ <h3 id="sec.identity-verify-assertion">Verifying Identity Assertions</h3>
-        <code><a href=
-        "#widl-RTCPeerConnection-peerIdentity">peerIdentity</a></code>
-        attribute and raises a simple event named <a href=
-        "#event-peeridentity">peeridentity</a> at itself. The assertion information to be
-        displayed MUST contain the domain name of the IdP as provided in the
-        assertion.
+        <li>
+          <p>The <code>RTCPeerConnection</code> resolves the <code><a href=
+          "#widl-RTCPeerConnection-peerIdentity">peerIdentity</a></code>
+          attribute with a new instance of <code>RTCIdentityAssertion</code>
+          that includes the IdP domain and peer identity.</p>
         </li>
 
-        <li>The browser MAY display identity information to a user in browser
-        UI. Any user identity information that is displayed in this fashion
-        MUST use a mechanism that cannot be spoofed by content.</li>
+        <li>
+          <p>The browser MAY display identity information to a user in browser
+          UI. Any user identity information that is displayed in this fashion
+          MUST use a mechanism that cannot be spoofed by content.</p>
+        </li>
       </ol>
 
-      <p>The IdP might fail to validate the identity assertion by providing an
-      "ERROR" response to the validation request. Validation can also fail due
-      to the additional checks performed by the browser. In both cases, the
-      process terminates and no identity information is exposed to the
-      application or the user.</p>
+      <p>This process can fail at any step above.  If identity validation fails,
+      the <code><a href=
+      "#widl-RTCPeerConnection-peerIdentity">peerIdentity</a></code> promise is
+      rejected with a <code>DOMError</code> that has a name of
+      <code>IdpError</code>.</p>
 
-      <p>The browser MUST raise an <code><a href=
-      "#event-idpvalidationerror">idpvalidationerror</a></code> event if
-      validation of an identity assertion fails for any reason.</p>
+      <p>If identity validation fails and there is
+      a <a href="#target-peer-identity">target peer identity</a> for the
+      <code>RTCPeerConnection</code>, the promise returned by
+      <code>setRemoteDescription</code> MUST be rejected.</p>
 
-      <p>If the "peerIdentity" constraint is applied to the
-      <code>RTCPeerConnection</code>, any error MUST cause <code><a href=
-      "#dom-peerconnection-setremotedescription">setRemoteDescription</a></code>
-      to fail.</p>
+      <p>If identity validation fails and there is no a <a
+      href="#target-peer-identity">target peer identity</a>, the value of the
+      <code><a href=
+      "#widl-RTCPeerConnection-peerIdentity">peerIdentity</a></code> MUST be set
+      to a new, unresolved promise instance.  This permits the use of
+      renegotiation (or a subsequent answer, if the session description was a
+      provisional answer) to resolve or reject the identity.</p>
```

Comment:
I don't quite understand how/where in the algorithm this new unresolved promise eventually gets resolved one way or another.  Can you briefly point me to what I missed in the section?